jdaviz/configs/default/plugins/plot_options/plot_options.py

```diff
@@ -67,29 +67,38 @@ def __init__(self):
 
         # Default x, y values(0-1) range chosen for a typical initial spline shape.
         # Can be modified if required.
-        self.update_knots(
-            x=np.array([0, 0.1, 0.2, 0.7, 1]),
-            y=np.array([0, 0.05, 0.3, 0.9, 1])
-        )
+        self._x = np.array([0, 0.1, 0.2, 0.7, 1])
+        self._y = np.array([0, 0.05, 0.3, 0.9, 1])
+        self.update_knots(self._x, self._y)
+
+    @property
+    def knots(self):
+        return (self._x, self._y)
+
+    @knots.setter
+    def knots(self, value):
+        x, y = value
+        if len(x) != len(y):
+            # Silently return
+            return
+        self.update_knots(x, y)
 
     def __call__(self, values, out=None, clip=False):
         # For our uses, we can ignore `out` and `clip`, but those would need
         # to be implemented before contributing this class upstream.
         return self.spline(values)
 
     def update_knots(self, x, y):
-        if len(x) != len(y):
-            raise ValueError("x and y must be the same length.")
-        self.x = x
-        self.y = y
+        self._x = x
+        self._y = y
         self.spline = make_interp_spline(
-            self.x, self.y, k=self.k, t=self.t, bc_type=self.bc_type
+            self._x, self._y, k=self.k, t=self.t, bc_type=self.bc_type
         )

```

Comment:
One downside to using `make_interp_spline` is that the resulting curve may not be monotonic in between knots. This can lead to singularities as knots get close to one another, and make silly colormappings.

We could use the PCHIP interpolation method instead to get monotonic behavior between knots, by changing roughly one line in plot options. Here's the difference for a simple set of knots:

```python
from scipy.interpolate import make_interp_spline, PchipInterpolator

x = [0, 0.4, 0.6, 1.0]
y = [0, 0.2, 0.6, 0.7]
t = np.linspace(0, 1, 1000)

# this is what we currently use:
bsp = make_interp_spline(x, y)(t)

# this is a simple-to-use alternative:
monotonic = PchipInterpolator(x, y)(t)

plt.scatter(x, y)
plt.plot(t, bsp, label='B Spline')
plt.plot(t, monotonic, label='PCHIP')
plt.legend()
plt.gca().set(xlabel='x', ylabel='y')
```
![pchip](https://github.com/spacetelescope/jdaviz/assets/3497584/945412ca-ff4c-4bdf-a327-c20ce1849215)
Here's the performance for a more problematic set of knots: 
```python
x = [0, 0.4, 0.5, 0.52, 1.0]
y = [0, 0.2, 0.6, 0.8, 0.85]
...
```
![pchip (1)](https://github.com/spacetelescope/jdaviz/assets/3497584/39191a43-0146-4873-b61e-ce3cb76af831)

Reply:
POs @camipacifici, @PatrickOgle – does this seem like a good idea to you?

Reply:
Thanks a lot, @bmorris3 for your suggestion about using PchipInterpolator. I looked into it, and you're right - it is a great fit for our needs, especially for keeping things smooth between knots. I'm going to update the class to use this. I appreciate your feedback. I also confirmed this with @camipacifici. @camipacifici, do you have anything to add to this?

Reply:
It looks great!!

Reply:
@kecnry, Thanks for bringing that up. With the new approach, the curve's behavior stays mostly within this range. However, PchipInterpolation doesn't automatically limit it to 0-1. I am looking into whether we need additional logic to enforce this range strictly. Any ideas?

Reply:
If it enforces monotonic in the curve itself and _we_ enforce that the "knots" can't extend outside that range (doesn't currently seem to be the case), then I think that should do the trick and we can remove the logic that replaces anything outside that range with 1s and 0s respectively.